frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -260,38 +260,40 @@ function filemanagerutil.genShowFolderButton(file, caller_callback, button_disab
     }
 end
 
-function filemanagerutil.genBookInformationButton(file, caller_callback, button_disabled)
+function filemanagerutil.genBookInformationButton(file, bookinfo, caller_callback, button_disabled)
     return {
         text = _("Book information"),
-        id = "book_information", -- used by covermenu
         enabled = not button_disabled,
         callback = function()
             caller_callback()
-            require("apps/filemanager/filemanagerbookinfo"):show(file)
+            local FileManagerBookInfo = require("apps/filemanager/filemanagerbookinfo")
+            FileManagerBookInfo:show(file, bookinfo and FileManagerBookInfo.extendProps(bookinfo))
         end,
     }
 end
 
-function filemanagerutil.genBookCoverButton(file, caller_callback, button_disabled)
+function filemanagerutil.genBookCoverButton(file, bookinfo, caller_callback, button_disabled)
+    local has_cover = bookinfo and bookinfo.has_cover
     return {
         text = _("Book cover"),
-        id = "book_cover", -- used by covermenu
-        enabled = not button_disabled,
+        enabled = not button_disabled and (not bookinfo or has_cover) and true or false,
         callback = function()
             caller_callback()
-            require("apps/filemanager/filemanagerbookinfo"):onShowBookCover(file)
+            local FileManagerBookInfo = require("apps/filemanager/filemanagerbookinfo")
+            FileManagerBookInfo:onShowBookCover(file)
         end,
     }
 end
 
-function filemanagerutil.genBookDescriptionButton(file, caller_callback, button_disabled)
+function filemanagerutil.genBookDescriptionButton(file, bookinfo, caller_callback, button_disabled)
+    local description = bookinfo and bookinfo.description
     return {
         text = _("Book description"),
-        id = "book_description", -- used by covermenu
-        enabled = not button_disabled,
+        enabled = not (button_disabled or bookinfo) or description and true or false,
```

Comment:
A bit hard to read this logic and guess the result of that. Could some parens help ?

I was trying to guess if all of this works when in classic mode and the coverbrowser plugin is disabled.
Can I assume all this work in that case and that these buttons are enabled, and it will go open the book to get metadata?

Reply:
Yes, if the coverbrowser is disabled, the book is opened to get the requested info.

Reply:
> A bit hard to read this logic

Yes it is because the logic is tricky.
`button_disabled == true` for deleted files. But if we have (remained) description in cache, the button must be enabled. If there is bookinfo but there is no description, the button must be disabled. But if the file is supported and there is no bookinfo (yet), the button must be enabled.

Reply:
Could that or bits of that be put in a comment ?

Also, can some parens help ?
At minima, instead of `this or that and true or false` (which is already hard to guess what the result is):
`(this or that) and true or false`